### PR TITLE
Add example promotion workflows

### PR DESCRIPTION
**Pull request by techpink**

### CODE REVIEW

File: examples/github/workflows/patcher.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Patcher
```
Comment (brikis98) [inline examples/github/workflows/patcher.yml:1]: Is this an example, or an action we want to publish?

File: examples/workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
+The promotional workflow uses 5 GitHub Actions workflows: 
+- `update-dev.yml` - this is the *Update Dev Dependencies* workflow, this workflow can be run periodically or tiggered by a release in an upstream infrastructure module repo
+- `update-stage.yml` - this is the *Update Stage Dependencies* workflow
+- `update-prod.yml` - this is the *Update Prod Dependencies* workflow
```
Comment (brikis98) [inline examples/workflows/README.md:5]: Other than dev, stage, and prod as the targets, are there any differences between these?

File: examples/workflows/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- `patcher.yml` - this is the *Patcher* workflow that prevents a PR being merged if it contains outstanding `README-TO-COMPLETE-UPDATE.md` files
+- `labeler.yml` - this is the *Labeler* workflow that ensures that the PRs are correctly labelled
+
+## The Updates Dependencies Workflows
+
+### Config
+The workflows require a GH personal access token to be defined using a secret named `PAT`.
```
Comment (brikis98) [inline examples/workflows/README.md:12]: What permissions does that PAT need?

**Review by brikis98 — COMMENTED** [review]
I'm a bit confused by the file layout... 

- Aren't we supposed to have an `action.yml` in the root of the repo to publish a GitHub Action?
- The folder structure is a bit odd: `examples/workflows` and `examples/github/workflows`.. What is the thought process behind it?
- We need a root README too. That could explain some of the above, of course.
- We need a license file. Please use Apache 2.0.

**Comment by techpink** [comment]
> * The folder structure is a bit odd: `examples/workflows` and `examples/github/workflows`.. What is the thought process behind it?

We decided to use the same repo for both the GHA workflows and the GHA itself. This is the PR for the GHA workflows. The action will be in a separate PR

File: examples/github/workflows/patcher.yml (hunk 1 above)
Comment (techpink) [reply examples/github/workflows/patcher.yml:1]: It's the patcher workflow, not the action

File: examples/github/workflows/patcher.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        uses: andstor/file-existence-action@v1
+        with:
+          files: "**/README-TO-COMPLETE-UPDATE.md"
+
+      - name: File exists
+        if: steps.check_files.outputs.files_exists == 'true'
+        run: |
+          echo "This pull request contains `README-TO-COMPLETE-UPDATE.md` files"
+          echo "Check for outstanding `README-TO-COMPLETE-UPDATE.md` files"
+          echo "This check will pass, when all the outstanding files have been actioned and deleted
```
[the remark this answers is not in the document]
Comment (techpink) [reply examples/github/workflows/patcher.yml]: Fixed. The job has been running OK without it

File: examples/github/workflows/update-dev.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
+name: Update Dev Dependencies
```
[the remark this answers is not in the document]
Comment (techpink) [reply examples/github/workflows/update-dev.yml:1]: These are the GHA workflows, not actions. The action will replace the run patcher and format results steps in the update-env job

File: examples/workflows/README.md (hunk 2 above)
Comment (techpink) [reply examples/workflows/README.md]: The events they listen for and the events that they send

File: examples/workflows/README.md (hunk 3 above)
Comment (techpink) [reply examples/workflows/README.md]: I've added additional context

**Review by marinalimeira — APPROVED** [review]

File: LICENSE.txt [hunk 6]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (brikis98) [inline LICENSE.txt:1]: Apache license please!

**Review by brikis98 — COMMENTED** [review]
Understood that these are the example workflows and the Action itself will come in a follow-up PR. However, I thought those example workflows were going to _use_ that very action... And as a result, most of the logic would end up in the action (something we maintain) rather than the example workflows (which customers would have to adapt/maintain). Maybe that'll all be updated once we have the action?

New folder structure makes more sense to me!

We still want a root README on this. OK to add in follow-up PR.

File: LICENSE.txt (hunk 6 above)
Comment (techpink) [reply LICENSE.txt]: Changed

**Comment by techpink** [comment]
There is a ticket in progress to publish the action - PATCHER-205

**Review by brikis98 — APPROVED** [review]
LGTM, assuming we don't forget a root README in follow up PRs :)

**Pull request merged**